api/src/org/apache/cloudstack/api/command/admin/host/ListHostsCmd.java

```diff
@@ -203,9 +201,22 @@ public void execute() {
                 hostResponse.setObjectName("host");
                 hostResponses.add(hostResponse);
             }
-
             response.setResponses(hostResponses, result.second());
         }
+        // Remove sensitive details from host response
+        List<HostResponse> hostResponsesList = response.getResponses();
+        for (HostResponse hostResponse: hostResponsesList) {
+            Map<String, String> hostDetails = hostResponse.getDetails();
+            if (hostDetails == null) continue;
+            if (hostDetails.containsKey("username")) {
+                hostDetails.remove("username");
+            }
+            if (hostDetails.containsKey("password")) {
+                hostDetails.remove("password");
+            }
+            hostResponse.setDetails(hostDetails);
```

Comment:
hey guys, its upto you how you want to handle it, our end goal is not to expose entire access details when listHosts api call is made. As mentioned earlier, it would be ideal to leave few characters for username and password VS doing hostDetails.remove.

Thanks
-ilya